optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/service/vehicle/VehicleService.java

```diff
@@ -53,32 +52,29 @@ public void createVehicle(VehicleData vehicleData) {
     }
 
     public void addVehicle(Vehicle vehicle) {
-        optimizer.addVehicle(Objects.requireNonNull(vehicle));
+        planner.addVehicle(Objects.requireNonNull(vehicle));
     }
 
     public void removeVehicle(long vehicleId) {
         Vehicle vehicle = vehicleRepository.removeVehicle(vehicleId);
-        optimizer.removeVehicle(vehicle);
+        planner.removeVehicle(vehicle);
     }
 
     public synchronized void removeAnyVehicle() {
         Optional<Vehicle> first = vehicleRepository.vehicles().stream().min(comparingLong(Vehicle::id));
         first.ifPresent(vehicle -> {
             Vehicle removed = vehicleRepository.removeVehicle(vehicle.id());
-            optimizer.removeVehicle(removed);
+            planner.removeVehicle(removed);
         });
```

Comment:
If it is `anyVehicle`, why is there the `min`? You should be able to do `findAny()` instead.

Reply:
`anyVehicle` == message to the caller telling that it's not guaranteed which vehicle will be deleted. `min` is an implementation choice to make the behavior deterministic.

Reply:
in that case, I suggest `findFirst()`, or at least documenting why the comparisons are being done.